chingu-voyages/v51-tier3-team-33
# Code Refactor - separated global context store with data fetching code

# Description
This was an update to the UI. The User context api was initially designed to handle both data storage, and data fetching. This commit, refactors that, by having a context for client side storage `UserContext` and another context api for data fetching `AppContextDataProvider`.
Since the changes were initially cloned from the repo made by [Olga's - v51_tier3_team33 repo fork](https://github.com/cvtqx/ExpenseSplitter), there have been some changes in the UI; mainly visible is the changes in the main layout.

## Main changes explained:
- commented out data fetching code in the User Context;
- modified the UserContext, to have an initial context object
- modified the UserContextProvider to utilize memoization (useMemo)
- Created a AppContextData context with initial null context
- Created an AppContextDataProvider to fetch data relevant to the entire app ui.
- refactored all providers wrapping the main layout.tsx into one global provider.

## How to test:
1. check into current branch
2. do `pnpm install` or `npm install` and `pnpm dev` or  `npm run dev` to run this PR locally

## Note:
Enjoy testing that app still works as it did prior.
Remember to include relevant .env file, if unavailable, in the app login as guest.
main purpose of this PR was to do code refactoring, to meet upto date react coding standards.
